ego4d/internal/university_files.py

```diff
@@ -0,0 +1,721 @@
+                AuxiliaryVideoComponentDataFile,
+                "university_video_id",
+                list_per_key=True,
+            )
+
+        # Load participant.csv, if available
+        file_name = "participant.csv"
+        participant_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            participant_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                Particpant,
+                "participant_id",
+            )
+
+        # Load synchronized_videos.csv, if available
+        file_name = "synchronized_videos.csv"
+        synchronized_video_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            synchronized_video_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                SynchronizedVideos,
+                "video_grouping_id",
+            )
+
+            # Load physical_setting.csv, if available
+        file_name = "physical_setting.csv"
+        physical_setting_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            physical_setting_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                PhysicalSetting,
+                "setting_id",
+            )
+
+            # Load annotations.csv, if available
+        file_name = "annotations.csv"
+        annotations_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            annotations_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                Annotations,
+                "university_video_id",
+            )
+
+    return (
+        video_metadata_dict,
+        video_components_dict,
+        auxiliary_video_component_dict,
+        participant_dict,
+        synchronized_video_dict,
+        physical_setting_dict,
+        annotations_dict,
+    )
+
+def validate_university_files(  # noqa :C901
```

Comment:
This function would be clearer if we refactored it by partitioning it into seperate functions, e.g. like so:

```
def validate_university_files(...):
    _validate_video_metadata(...)
    _validate_syncronized_video(...)
    _validate_mp4_files(...)
    # etc.
```

where `_validate_video_metadata` accepted the already loaded data. In this way we can write some unit tests for each component.

Right now it's a bit hard to read and will be quite annoying to test.

---

Additionally, when writing each function, it would be good to have a docstring for the list of properties we are checking, e.g.

````
def _validate_mp4_files(...):
    """
    This function checks a set of MP4 files on S3 for the following properties:
    - null fps
    - consistent width/height
    - etc.
    """
```